File: server/src/models/experience.rs
use serde::Serialize;
use sqlx::FromRow;

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct ExperienceModel {
    pub id: i32,
    pub user_id: i32,
    pub company: String,
    pub title: String,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub exp_type: Option<String>,
    pub description: Option<String>,
    pub created_at: String,
}

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct PublicExperienceModel {
    pub company: String,
    pub title: String,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub exp_type: Option<String>,
    pub description: Option<String>,
}

File: src/types/experience.ts
export interface ExperienceModel {
  id: number;
  user_id: number;
  company: string;
  title: string;
  start_date?: string;
  end_date?: string;
  exp_type?: string;
  description?: string;
  created_at: string;
}

export interface PublicExperienceModel {
  company: string;
  title: string;
  start_date?: string;
  end_date?: string;
  exp_type?: string;
  description?: string;
}

export interface AuthExperienceModel extends PublicExperienceModel {
  id?: number;
}

export interface AddExperiencePayload {
  company: string;
  title: string;
  start_date?: string;
  end_date?: string;
  exp_type?: string;
  description?: string;
}
